src/app/calendar-app/runbox-calendar-event.ts

```diff
@@ -69,137 +88,437 @@ export class RunboxCalendarEvent implements CalendarEvent {
+    }
+
+    set recursByMonthDay(value: string[]) {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        recur.setComponent('bymonthday', value);
+        this.event.component.updatePropertyWithValue('rrule', recur);
+    }
+
+    get recursByYearDay(): string[] {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        if (!recur) {
+            return [];
+        }
+        const byyeardays = recur.getComponent('byyearday');
+        return byyeardays.map((day) => day.toString());
+    }
+
+    set recursByYearDay(value: string[]) {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        recur.setComponent('byyearday', value);
+        this.event.component.updatePropertyWithValue('rrule', recur);
+    }
+
+    // Jan-Dec numbered 1-12
+    get recursByMonth(): string[] {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        if (!recur) {
+            return [];
+        }
+        const bymonth = recur.getComponent('bymonth');
+        return bymonth.map((month) => month.toString());
+    }
+
+    set recursByMonth(value: string[]) {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        recur.setComponent('bymonth', value);
+        this.event.component.updatePropertyWithValue('rrule', recur);
+    }
+
     set allDay(value) {
-        this.event.startDate.isDate = value;
-        if (this.event.endDate) {
-            this.event.endDate.isDate = value;
+        this._dtstart.isDate = value;
+        if (this._dtend) {
+            this._dtend.isDate = value;
         }
         // I know this looks silly, but without this
         // ICAL.Event will not notice the isDate change
         // and the event metadata will still be off
-        this.event.startDate = this.event.startDate;
-        this.event.endDate   = this.event.endDate;
+        // FIXME: where is this set/used?
+        // this.event.startDate = this._dtstart;
+        // this.event.endDate   = this._dtend;
     }
 
-    static fromIcal(id: string, ical: string): RunboxCalendarEvent {
-        return new this(id, ICAL.parse(ical));
+    get isException(): boolean {
+        return this.event.component.hasProperty('recurrence-id');
     }
 
     // creates an unnamed event for today
     static newEmpty(): RunboxCalendarEvent {
         return new this(
             undefined,
-            [ 'vcalendar', [], [
-                [ 'vevent',
-                    [
-                        [ 'dtend',   {}, 'date-time', '2019-10-02T14:00:00' ],
-                        [ 'dtstart', {}, 'date-time', '2019-10-02T12:00:00' ],
-                        [ 'summary', {}, 'text', '' ] ],
-                    []
-                ]
-            ] ]
+            new ICAL.Event(new ICAL.Component('vevent')),
+            ICAL.Time.fromData({year: 2019, month: 10, day: 2, hour: 14, minute: 0}),
```

Comment:
Maybe we can take this opportunity to actually follow the comment and create that even for today ;)

Reply:
Crazy idea.. ;)
Luckily ICAL.Time has a function for that..